client/state/selectors/get-current-plan-purchase-id.js

```diff
@@ -0,0 +1,33 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { get, isNaN } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getCurrentPlan } from 'state/sites/plans/selectors';
+
+/**
+ * Returns a site's current plan purchase ID or null if the site doesn't exist or the purchase ID
+ * is unknown.
+ *
+ * The ID returned by this selector is retrieved from the site plan. It is intended to provide a
+ * means of retrieving full purhcase information based on a site and it's plan information.
+ * Caution! It _does not_ retrieve the ID from a purchase.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @return {?Number}        Purchase ID if known
+ */
+export default function getCurrentPlanPurchaseId( state, siteId ) {
+	const result = get( getCurrentPlan( state, siteId ), 'id', null );
+
+	// getCurrentPlan uses an "assembler" which may have NaN in the `id`.
+	if ( isNaN( result ) ) {
+		return null;
```

Comment:
> Since we probably know the ID, wouldn't it make sense to use it? It would simply be `result.id` here, correct?

I don't follow.

We don't know the ID, that's exactly the problem here. [Here](https://github.com/Automattic/wp-calypso/blob/887cd5b4e094f7c235de8cbd833604a9d268ef60/client/state/sites/plans/assembler.js#L25) you may get `NaN`:

```js
const createSitePlanObject = ( plan ) => {
  if ( ! plan ) {
    return {};
  }

  return {
    id: Number( plan.id ),  // <-- Might be a NaN   >:(
    /* ... */
  }
}
```

I ran into this in [this test](https://github.com/Automattic/wp-calypso/blob/2ee87dd2fa3efba4a63591b15a12e34cfd08e672/client/state/selectors/test/get-current-plan-purchase-id.js#L62-L80) I believe while working on this.